File: 04-10-2004_ivide-players-into-teams-of-equal-skill.cpp
class Solution {
public:
    long long dividePlayers(vector<int>& skill) 
    {
        unordered_map<int,int> mp;
        int mini=INT_MAX,maxi=INT_MIN;
        for(auto& i:skill)
        {
           mp[i]++;
           mini=min(mini,i);
           maxi=max(maxi,i);
        }
        long long ans=0,sum=mini+maxi,n=skill.size();
        for(auto& i:mp)
        {
            if(i.second==0) continue;
            int rem=sum-i.first;
            if(i.second!=mp[rem]) return -1;
            else
            {
                if(rem==i.first) i.second/=2;
                ans+=(long long) i.second*rem*i.first;
                mp[rem]=0;
            }

        }
    return ans;  
    }
}; 

File: 04-10-2004_ivide-players-into-teams-of-equal-skill.py
from collections import defaultdict

class Solution:
    def dividePlayers(self, skill: list[int]) -> int:
        mp = defaultdict(int)
        mini = float('inf')
        maxi = float('-inf')
        
        for i in skill:
            mp[i] += 1
            mini = min(mini, i)
            maxi = max(maxi, i)
        
        ans = 0
        total_sum = mini + maxi
        n = len(skill)
        
        for i in list(mp.keys()):
            if mp[i] == 0:
                continue
            
            rem = total_sum - i
            if rem not in mp or mp[rem] != mp[i]:
                return -1
            else:
                if rem == i:
                    mp[i] //= 2
                ans += mp[i] * rem * i
                mp[rem] = 0
        
        return ans
